docs/specs/ISA.md

```diff
@@ -1,90 +1,100 @@
+There is a single special purpose register `pc` for the program counter of type `F` which stores the location of the instruction being executed. (We may extend `pc` to multiple field elements to increase the program address space size in the future.)
 
 The program code is committed as a cached trace. The validity of the program code and its cached trace must be checked
 outside of ZK. A valid program code must have all instructions stored at locations in range `[0, 2^PC_BITS)`. While the
 instructions can be stored at any locations, we will by default follow RISC-V in storing instructions at multiples of
 `DEFAULT_PC_STEP = 4`.
 
-## Memory
+### Memory
 
-Memory is comprised of addressable cells, each cell containing a single field element.
-Instructions of the VM may access (read or write) memory
-as single cells or as a contiguous list of cells. Such a contiguous list is called a _block_, and
-a memory access (read/write) to a block is a _block access_.
+Memory is comprised of addressable cells indexed by **address space** and **pointer**. Each cell is a single field element, and VM instructions may access (read or write) memory
+as single cells or as a contiguous list of cells in a single address space. Such a contiguous list is called a **block**, and
+a memory access (read/write) to a block is a **block access**.
 The architecture distinguishes between block accesses of different sizes as this has significant performance
 implications.
-The number of cells in a block access is restricted to powers of two, of which the following are supported: 1, 2, 4, 8,
+The number of cells in a block access is restricted to powers of two in: 1, 2, 4, 8,
 16, 32, 64. Block accesses do not need to be
 aligned, i.e., a block access of size $N$ can start from a pointer with value not dividing $N$ (as an integer).
 
-We also leave open the possibility in the future that different address spaces (see below) can be dedicated to handling
+We leave open the future possibility that different address spaces (see below) can be dedicated to handling
 data with certain block sizes, effectively declaring a word-size for that address space, but this is not currently
-implemented. At present there are two types of blocks we have in mind
+implemented. At present, there are two types of blocks used in OpenVM:
 
 - **[FVEC]** A block consisting of `[F; N]` arbitrary field elements.
 - **[LIMB]** A block consisting of `[F; N]` field elements, where each field element has as its canonical representation
-  a limb in `[0, 2^LIMB_BITS)`. This would emulate a word in RISC-V memory.
+  a limb in `[0, 2^LIMB_BITS)`. This is used to emulate a word in RISC-V memory.
 
 While not relevant to the ISA itself, the ZK circuit implementation does usually represent a block `[F; N]` as `N`
 contiguous field elements in the same row of the trace matrix.
 
-## Immediate Values
+### Immediate Values
```

Comment:
This was old; IMO immediate values and registers should only be mentioned in the risc-v section